core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMaterializedViewDefinition.java

```diff
@@ -35,10 +37,28 @@
     private final Optional<String> owner;
     private final Map<String, Object> properties;
 
+    public ConnectorMaterializedViewDefinition(
+            @JsonProperty("originalSql") String originalSql,
+            @JsonProperty("storageTable") Optional<String> storageTable,
+            @JsonProperty("catalog") Optional<String> catalog,
+            @JsonProperty("schema") Optional<String> schema,
+            @JsonProperty("columns") List<Column> columns,
+            @JsonProperty("comment") Optional<String> comment,
+            @JsonProperty("owner") Optional<String> owner,
+            @JsonProperty("properties") Map<String, Object> properties)
+    {
+        this(originalSql, requireNonNull(storageTable, "storageTable is null").orElse(null), catalog, schema, columns, comment, owner, properties);
+    }
+
+    /*
+     * This constructor is for JSON deserialization only. Do not use.
+     * It's marked as @Deprecated to help avoid usage, and not because we plan to remove it.
```

Comment:
Maybe we could remove it in the future, but I'm not sure if @Nullable vs Optional are compatible during deserialization

Reply:
can you check that maybe?

Reply:
It can, if we use `@JsonSetter(nulls = Nulls.AS_EMPTY)`
https://stackoverflow.com/questions/32398885/jackson-deserializer-change-null-collection-to-empty-one 
Will update comment